src/linear_algebra/basic.lean

```diff
@@ -300,6 +300,13 @@ instance {T : Type*} [monoid T] [distrib_mul_action T M₂] [smul_comm_class R T
   smul_comm_class S T (M →ₗ[R] M₂) :=
 ⟨λ a b f, ext $ λ x, smul_comm _ _ _⟩
 
+-- example application of this instance: if S and R are comm_semirings and R is an S-algebra
+-- then the S-actions and R-actions on Hom_R(M,M₂) coincide.
```

Comment:
This comment should probably mention `T` - or just be removed

Reply:
The comment is supposed to orient a mathematician looking at all this twaddle and thinking "what the heck is the actual point of this"? From a CS point of view these instances we're coming up with are probably exactly the right ones, but probably no mathematician will ever use it in the generality in which it's written. I'll rephrase to mention T.

Reply:
If its for mathematicians, it probably belongs in a doc-comment so it appears in doc-gen

Reply:
We could always use more in the docs along the lines of "here's an example on how to use this code"! I think we really only have text like that for tactics, but it's really helpful for other stuff as well too. (The trouble would be keeping it up to date since things are being refactored constantly.)